src/screens/DiscoveryScreenV2/elements/DiscoveryItemList.js

```diff
@@ -78,25 +79,50 @@ const DomainList = (props) => {
     );
   };
 
+  const renderProfilePicture = () => {
+    if (item.karmaScore) {
+      return (
+        <CircleGradient
+          testId="images"
+          fill={item.karmaScore}
+          size={normalize(51)}
+          width={normalize(2.3)}>
+          <Image
+            testId="images"
+            source={{
+              uri: item.image
+            }}
+            style={styles.profilepicture}
+            width={48}
+            height={48}
+          />
+        </CircleGradient>
+      );
+    }
+    return (
+      <Image
+        testId="images"
+        source={{
+          uri: item.image
+        }}
+        style={styles.profilepicture}
+        width={48}
+        height={48}
+      />
+    );
+  };
+
   return (
     <View style={styles.container}>
       <TouchableOpacity
         testID="pressbody"
         onPress={() => onPressBody(item)}
         style={styles.wrapProfile}>
-        <View style={styles.card}>
+        <View style={styles.card} testID="images">
           {!isHashtag ? (
             <React.Fragment>
               {item.image && typeof item.image === 'string' && item.image.length > 0 ? (
-                <Image
-                  testID="images"
-                  source={{
-                    uri: item.image
-                  }}
-                  style={styles.profilepicture}
-                  width={48}
-                  height={48}
-                />
+                renderProfilePicture()
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `renderProfilePicture` function is now being called within the render method of the component, replacing the previous inline rendering of the image. This change seems to be correct and should not introduce any issues.
```diff
-                {item.image && typeof item.image === 'string' && item.image.length > 0 ? (
-                 <Image
-                   testID="images"
-                   source={{
-                     uri: item.image
-                   }}
-                   style={styles.profilepicture}
-                   width={48}
-                   height={48}
-                 />
-                ) : (
-                  <View testID="noimage">{renderDefaultImage(DefaultImage)}</View>
-                )}
+                {item.image && typeof item.image === 'string' && item.image.length > 0 ? (
+                 renderProfilePicture()
+                ) : (
+                  <View testID="noimage">{renderDefaultImage(DefaultImage)}</View>
+                )}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->